### PR TITLE
feat(core): add dynamic queries schematic

### PR DESCRIPTION
**Pull request by crisbeto**
Adds a schematic that will remove the explicit `static: false` flag from dynamic queries. E.g.

#### Before
```ts
import { Directive, ViewChild, ContentChild, ElementRef } from '@angular/core';

@Directive()
export class MyDirective {
  @ViewChild('child', { static: false }) child: any;
  @ViewChild('secondChild', { read: ElementRef, static: false }) secondChild: ElementRef;
  @ContentChild('thirdChild', { static: false }) thirdChild: any;
}
```

#### After
```ts
import { Directive, ViewChild, ContentChild, ElementRef } from '@angular/core';

@Directive()
export class MyDirective {
  @ViewChild('child') child: any;
  @ViewChild('secondChild', { read: ElementRef }) secondChild: ElementRef;
  @ContentChild('thirdChild') thirdChild: any;
}
```

### CODE REVIEW

**Review by devversion — APPROVED** [review]
LGTM 🎉 

Just to mention one edge case: If someone has the query options set through a variable declaration, the migration will currently do nothing. It could show a message though that the given query should be checked manually. Just mentioning this as it could make the migration more robust.

<sup>Not saying we should do it; just bringing this up for awareness; we only saw these type of queries rarely in the static-query migration<sup>.

**Review by kara — CHANGES_REQUESTED** [review]

**Comment by crisbeto** [comment]
The feedback has been addressed @kara.

File: packages/core/schematics/test/dynamic_queries_migration_spec.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        @ViewChild('child', { read: ElementRef, static: true }) child: ElementRef;
+      }
+    `);
+
+    await runMigration();
+    expect(tree.readContent('/index.ts'))
+        .toContain(`@ViewChild('child', { read: ElementRef, static: true }) child: ElementRef;`);
+  });
+
+  it('should not change dynamic ContentChild queries', async() => {
```
[the remark this answers is not in the document]
Comment (kara) [reply packages/core/schematics/test/dynamic_queries_migration_spec.ts]: Looks like this change got lost in the rebase? Still says dynamic

**Review by kara — APPROVED** [review]
LGTM, once last comment is addressed

File: packages/core/schematics/test/dynamic_queries_migration_spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (crisbeto) [reply packages/core/schematics/test/dynamic_queries_migration_spec.ts]: Sorry about that, I think I messed it up when I was switching computers. It's fixed now.

**Comment by crisbeto** [comment]
Rebased and addressed the last comment @kara.

**Comment by kara** [comment]
[presubmit](http://test/OCL:268249985:BASE:268250017:1568135250033:63fbd00f)

**Pull request closed without merging**